core/memory.c

```diff
@@ -102,3 +102,13 @@ static PyObject* memfree(PyObject *self, PyObject *args) {
 
     return PyLong_FromLong(0);
 }
+
+
+void* safe_malloc(size_t size, const char* var_name) {
+    void* ptr = malloc(size);
+    if (ptr == NULL) {
+        fprintf(stderr, "Memory allocation failed for %s\n", var_name);
+        exit(1);
+    }
+    return ptr;
+}
```

Comment:
it will stop but it will not crash, it will throw an exception error that we can handle, and show a message in the console so people can report the issue or continue the last process just ignore one particle. so basically the Blender will not crash, the simulation maybe.